plugins/search-react/src/components/SearchTracker/SearchTracker.tsx

```diff
@@ -17,20 +17,34 @@
 import React, { useEffect } from 'react';
 import { useAnalytics } from '@backstage/core-plugin-api';
 import { useSearch } from '../../context';
+import usePrevious from 'react-use/lib/usePrevious';
 
 /**
  * Capture search event on term change.
  */
 export const TrackSearch = ({ children }: { children: React.ReactChild }) => {
+  const useHasChanged = (value: any) => {
+    const previousVal = usePrevious(value);
+    return previousVal !== value;
+  };
+
   const analytics = useAnalytics();
-  const { term } = useSearch();
+  const { term, result } = useSearch();
+
+  const numberOfResults = result.value?.numberOfResults ?? undefined;
+
+  // Stops the analtyics event from firing before the new search engine response is returned
```

Comment:
Thanks for catching/fixing this!  Solution _seems_ reasonable to me.

I feel like this is sufficiently complex enough now that perhaps a test would be in order for the tracker component?  Just to make sure that we continue to only capture a single search event.

Reply:
We're already testing for this in the SearchBar test (SearchBar is the only component that uses SearchTracker)
https://github.com/backstage/backstage/blob/600810c3674a9b7e1df552130952e02a9cd98e79/plugins/search-react/src/components/SearchBar/SearchBar.test.tsx#L341

I had previously updated this test to assert that 2 analytics events were called (this was a mistake on my part)

Removing the `hasFinishedLoading` condition in the useEffect hook causes the test to fail (as we get 2 analytics events firing).

Reply:
Aaah, right!  I'd forgotten this gets tested more end-to-end than other "components."

Thanks, that makes sense.